x-pack/plugins/security_solution/server/endpoint/routes/actions/service.ts

```diff
@@ -94,12 +150,22 @@ export const getAuditLogResponse = async ({
     throw new Error(`Error fetching actions log for agent_id ${elasticAgentId}`);
   }
 
+  const resultData = result.body.hits.hits.map((e) => ({
+    type: e._index.startsWith('.fleet-actions') ? 'action' : 'response',
+    item: { id: e._id, data: e._source },
+  })) as ActivityLog['data'];
+
+  const endpointActionIds = resultData
+    .filter((e) => e.type === 'action')
+    .map((e) => e.item.data.action_id);
+
+  const data = resultData.filter((e) =>
+    e.type === 'response' ? endpointActionIds.includes(e.item.data.action_id) : e
+  );
+
```

Comment:
This can break pagination for the frontend. What if this query returns a (valid endpoint) response, where the (valid endpoint) action was part of the previous page? It would get filtered out.

Also what if we had this situation:

```
- Actions & responses x 10
- OSQuery x 500
- Actions & responses x 10
```

With the correct filters for actions, you will probably get all 20 actions in the first response page. But the responses for the last group will be 500 osquery-responses away. 

And how does the frontend know when its done? empty response, right? You would have a lot of empty response pages until you got to those results

Reply:
That makes sense. And the way the FE is setup the paging won't increment if the backend returns an empty list. We'll have to use a two query solution that you suggested.